document/core/syntax/modules.rst

```diff
@@ -239,29 +237,26 @@ starting with the smallest index not referencing a global :ref:`import <syntax-i
 Element Segments
 ~~~~~~~~~~~~~~~~
 
-The initial contents of a table is uninitialized. *Element segments* initialize a subrange of a table from a static :ref:`vector <syntax-vec>` of elements.
+The initial contents of a table is uninitialized. *Element segments* can be used to initialize a subrange of a table from a static :ref:`vector <syntax-vec>` of elements.
 
 Element segments can be *active* or *passive*. An active element segment copies its elements into a table during :ref:`instantiation <exec-instantiation>`. A passive element segment's elements can be copied using the |TABLEINIT| instruction.
 
-The |MELEM| component of a module defines a vector of active element segments. Each active element segment defines the |ETABLE| and the starting |EOFFSET| in that table to initialize.
+The |MELEM| component of a module defines a vector of element segments. Each active element segment defines the |ETABLE| and the starting |EOFFSET| in that table to initialize. Each passive element segment only defines its contents.
 
 .. math::
    \begin{array}{llll}
    \production{element segment} & \elem &::=&
+     \EACTIVE~\aelem ~|~
+     \EPASSIVE~\pelem \\
+   \production{active element segment} & \aelem &::=&
```

Comment:
Nit: I don't think introducing auxiliary non-terminals for these is needed, just inline them above.

Also, both cases can be distinguished by structure, so the explicit active//passive tags aren't really necessary (cf. function instances, for example). One annoyance with such tags is that they prevent you from writing e.g. \elem.\einit and instead force you to always pattern-match the tag away first.

Otherwise looks good!

Reply:
I liked having the tags since it made it clearer which record is used for active vs. passive, but the text explains it so it's probably OK. Agreed it will be nicer to use `\einit` directly.

Not quite sure what you mean by function instances -- where are they distinguished by structure?

Reply:
They can either be regular or host functions, which are different records, see [here](http://webassembly.github.io/spec/core/exec/runtime.html#function-instances).

Reply:
Ah, right. I was looking at functions not function instances. :-)

Reply:
@rossberg BTW, I've just started to work on validation but I think I need to store whether the segment is active or passive in the index. I considered using a bool for this, but it seems better to use an enumeration. Any suggestions?

Reply:
Yes, I'd use the "activeness" enumeration you had in the abstract syntax in an earlier version. Don't have any good suggestion for a better name, though.

Btw, this makes me wonder: in principle, we could allow importing/exporting passive segments. Has that possibility ever been discussed?